tensorflow/contrib/learn/python/learn/monitors.py

```diff
@@ -177,14 +182,26 @@ def set_estimator(self, estimator):
     def create_val_feed_dict(self, inp, out):
         """Set tensorflow placeholders and create validation data feed"""
         self.val_feeder.set_placeholders(inp, out)
-        self.val_dict = self.val_feeder.get_feed_dict_fn()()
+        # If mini batches, get a function else full dataset.
+        if self._batch_size > 0:
+            self.val_dict_fn = self.val_feeder.get_feed_dict_fn()
+        else:
+            self.val_dict = self.val_feeder.get_feed_dict_fn()()
```

Comment:
Why the distinction? Together with the branch in _set_last_loss_seen which always does the same thing in the end -- compute self.val_feeder.get_feed_dict_fn()(). 

Is val_dict used elsewhere?